### PR TITLE
ISSUE-229: Remove check consistency code, test and related docs

### PR DESCRIPTION
**Pull request by ezienecker**
# Description of change

Preparation for https://github.com/iotaledger/iota-java/issues/229

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which remove functionality)
- Documentation Fix

## How the change has been tested

run mvn clean install

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests using ginkgo that prove my fix is effective or that my feature works -> not needed
- [ ] New and existing unit tests pass locally with my changes -> not needed

### CODE REVIEW

File: docs/iota-java/isPromotable.md [hunk 1]
```diff
@@ -20,11 +20,6 @@ Checks if a transaction hash is promotable
 |:---------------|:--------|
 | [ArgumentException](https://github.com/iotaledger/iota-java/blob/master/jota/src/main/java/org/iota/jota/error/ArgumentException.java) | when we can't get the consistency of this transaction |
```
Comment (kwek20) [inline docs/iota-java/isPromotable.md]: Can you remove the exception here?

File: jota/src/main/java/org/iota/jota/IotaAPI.java [hunk 2]
```diff
@@ -1611,14 +1611,12 @@ public void validateTransfersAddresses(String seed, int security, List<String> t
      * @param tail the {@link Transaction} we want to promote
      * @return <tt>true</tt> if it is, otherwise <tt>false</tt>
      * @throws ArgumentException when we can't get the consistency of this transaction
-     * @see #checkConsistency(String...)
      */
     @Document
     public boolean isPromotable(Transaction tail) throws ArgumentException {
```
Comment (kwek20) [inline jota/src/main/java/org/iota/jota/IotaAPI.java:1640]: and this method doesnt throw it anymore

File: jota/src/main/java/org/iota/jota/IotaAPI.java [hunk 3]
```diff
@@ -1628,7 +1626,6 @@ public boolean isPromotable(Transaction tail) throws ArgumentException {
      * @return <tt>true</tt> if it is, otherwise <tt>false</tt>
      * @throws ArgumentException when we can't get the consistency of this transaction
```
Comment (kwek20) [inline jota/src/main/java/org/iota/jota/IotaAPI.java]: This method still throws it, but replace explanation to "When we cant find the transaction"

**Review by kwek20 — CHANGES_REQUESTED** [review]
Can you also remove `CheckConsistencyResponse.java`? Thanks for the fix!

**Comment by kwek20** [comment]
@ezienecker

**Comment by ezienecker** [comment]
@kwek20 done

**Comment by kwek20** [comment]
Legend!

**Pull request merged**